docs/reference.md

```diff
@@ -131,3 +131,57 @@ dedicated test-suite, consisting of several tests exercising various runtime
 behaviors, for each StableHLO Op. The tests can be found
 [here](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/) (e.g.
 interpret\_\*.mlir).
+
+### Testing guidelines
+
+**(G1) Do we need to test for all the supported types for every op ?**
+
+We can use the following thumb rule to decide it:
+
+While implementing an op, if we need to introduce code in the corresponding
+`eval` function to handle types differently, then it is imperative to have tests
+to cover for all those types. But in cases where a set of types are handled
+uniformly, like for `add` op, all the supported integer types are handled alike
+using `llvm::APInt` APIs, then we can skip adding tests for all those types and,
+instead, add a single test with a type, with maximum bit-width (an arbitrary
+choice), from the set.
+
+For quantization ops, like `uniform_dequantize`, `uniform_quantize`, the
+decision is TBD.
+
+**(G2) How about adding tests dedicated for testing the interpreter infrastructure ?**
+
+The interpreter infrastructure is mostly straightforward and can be added it in
+our trust base. The only non-trivial part is how various types are packed into
+and unpacked from the underlying interpreter storage. As discussed in (G1), we
+will be testing only those types of an op which are handled differently. With
+that is possible that the packing/un-packing code, corresponding to different
+variants of integer/floating point types, might not get fully covered during
+testing.
+
+To ensure that we can chose an op, like `add`, which supports all the StableHLO
+element types and write exhaustive tests. Fortunately, we already have it.
```

Comment:
How about instead we have a dedicated file for this, separate from `interpreter_add.mlir`? E.g. `interpreter_types.mlir` or something?

Reply:
Please see my reply above.